app/models/questionnaire.rb

```diff
@@ -6,6 +6,9 @@ class Questionnaire < ApplicationRecord
   validates :key, presence: true, uniqueness: true, format: { with: /\A[a-z]+[a-z_0-9]*\Z/ }
   serialize :content, Array
   validate :all_content_ids_unique
+  validate :all_questions_have_types
```

Comment:
Yep. This is because people that use the previewer sometimes still submit questionnaires that don't seed. So I had to add validations, and then do a Questionnaire.new(content: params[:content], ...).valid? And give the errors back to the previewer.